subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/SingleOriginIncrementalAnnotationProcessingIntegrationTest.groovy

```diff
@@ -16,22 +16,27 @@
 
 package org.gradle.api.tasks.compile
 
+import org.gradle.api.JavaVersion
 import org.gradle.api.internal.tasks.compile.processing.IncrementalAnnotationProcessorType
+import org.gradle.integtests.fixtures.AvailableJavaHomes
 import org.gradle.language.fixtures.HelperProcessorFixture
 import org.gradle.language.fixtures.NonIncrementalProcessorFixture
 import org.gradle.language.fixtures.ServiceRegistryProcessorFixture
+import org.gradle.util.TextUtil
 
 class SingleOriginIncrementalAnnotationProcessingIntegrationTest extends AbstractIncrementalAnnotationProcessingIntegrationTest {
+    private HelperProcessorFixture helperProcessor
```

Comment:
It might be good to have a scenario where there is a cycle between the set of generated and non-generated source, and see how well incremental compile works when things change.

Reply:
What do you mean by "how well it works"? Can you give me a small scenario and what you'd expect to happen?